lib/utils/typical/parser.go

```diff
@@ -56,20 +56,30 @@ type Expression[TEnv, TResult any] interface {
 }
 
 // ParserSpec defines a predicate language.
-type ParserSpec struct {
-	// Variables defines all of the literals and variables available to
+type ParserSpec[TEnv any] struct {
+	// Variables define the literals and variables available to
 	// expressions in the predicate language. It is a map from variable name to
 	// definition.
 	Variables map[string]Variable
 
-	// Functions defines all of the functions available to expressions in the
+	// Functions define the functions available to expressions in the
 	// predicate language.
 	Functions map[string]Function
 
-	// Methods defines all of the methods available to expressions in the
+	// Methods define the methods available to expressions in the
 	// predicate language. Methods are just functions that take their receiver
 	// as their first argument.
 	Methods map[string]Function
+
+	// GetUnknownIdentifier is used to retrieve any identifiers that cannot
+	// be determined statically. If not defined, any unknown identifiers result
+	// in a [UnknownIdentifierError].
+	//
+	// Useful in situations where a parser may allow specifying nested paths
+	// to variables without requiring users to enumerate all paths in [ParserSpec.Variables].
+	// Caution should be used when using this method as it shifts type safety
+	// guarantees from parse time to evaluation time.
```

Comment:
```suggestion
	// guarantees from parse time to evaluation time.
	// Typos in identifier names will also be caught at evaluation time instead of parse time.
```